### PR TITLE
[bdg-utils-13] Added code to implement a poisson mixture model.

### LINKED ISSUE #13: Add statistical model fitting code
We need to fit a mixture of Poisson distributions for https://github.com/bigdatagenomics/adam/issues/401.

### PR DESCRIPTION
**Pull request by fnothaft**
Adds code to fit a mixture of poisson random variables. Additionally, adds a few mathematical utils. Closes #13.

### CODE REVIEW

**Comment by fnothaft** [comment]
@laserson apologies for the delay here; I cut the Gaussian code out, and have rebased.

File: bdg-utils-misc/src/main/scala/org/bdgenomics/utils/misc/MathUtils.scala [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+  /**
+   * Aggregation function for arrays which does not allocate a new array.
+   *
+   * @param a1 Array to aggregate.
+   * @param a2 Array to aggregate.
+   * @return Overwrites a1 with the aggregated value, and returns a1.
+   */
+  def aggregateArray(a1: Array[Double], a2: Array[Double]): Array[Double] = {
+    (0 until a1.length).foreach(i => a1(i) += a2(i))
```
Comment (laserson) [inline bdg-utils-misc/src/main/scala/org/bdgenomics/utils/misc/MathUtils.scala]: Still need to remove this, @fnothaft?

Comment (fnothaft) [inline bdg-utils-misc/src/main/scala/org/bdgenomics/utils/misc/MathUtils.scala]: Ah, thanks for the ping! Missed this note.

Comment (fnothaft) [inline bdg-utils-misc/src/main/scala/org/bdgenomics/utils/misc/MathUtils.scala]: Actually, I lied earlier. This code is indeed still used in training the poisson mixture.

**Comment by fnothaft** [comment]
@laserson ping on merging.

**Pull request merged**